Backend app crashes if db connection unavailable

### Community Note

- Please vote on this issue by adding a 👍:+1:
  [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)
  to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions,
  they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a
  comment

<!--- Thank you for keeping this note for the community --->

### App Version

<!--- Please list the release you are using. If you are not running the latest release, please upgrade
because your issue may have already been fixed. --->

### App Configuration File

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```yaml
# Copy-paste your config.yaml here
```

### Panic Output

```
2021/08/13 01:08:39 Unable to connect to Gremlin database at: dial tcp 127.0.0.1:8182: connect: connection refused
2021/08/13 01:08:39 dial tcp 127.0.0.1:8182: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa1d746]

goroutine 1 [running]:
main.reIndex(0x0, 0x0, 0xc000010110, 0x1, 0x1, 0x0, 0xc7d6e0)
        /code/graph.go:10 +0x26
main.main()
        /code/main.go:36 +0x2a6
```

### Expected Behavior

<!--- What should have happened? --->

App should continue and log that DB is unavailable. Potentially surfacing this through to UI

### Actual Behavior

<!--- What actually happened? --->
App panics

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->
Start backend app without DB or invalid DB connection string

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

- 0000

--- a/app/database/gremlin.go
+++ b/app/database/gremlin.go
@@ -34,7 +34,7 @@ func newGremlin(dsn string) (Graph, error) {
 		conn, err := grammes.DialWithWebSocket(dsn, grammes.WithErrorChannel(errs))
 		if err != nil {
 			if retryCount == 0 {
-				log.Println("Unable to connect to Gremlin database at: " + err.Error())
+				log.Println("Unable to connect to Gremlin database at: " + dsn)
 				return &Gremlin{}, err
 			}
 

--- a/app/graph.go
+++ b/app/graph.go
@@ -65,3 +65,7 @@ func loadNode(graph database.Graph, node *connectors.Node) (*database.Entity, er
 	}
 	return entityFrom, nil
 }
+
+func graphConnectionErr() error {
+	return fmt.Errorf("connection error: database unavailable")
+}

--- a/app/graphql.go
+++ b/app/graphql.go
@@ -11,6 +11,9 @@ import (
 func resolveRelationships(id interface{}, p graphql.ResolveParams) ([]*database.Entity, error) {
 	context, ok := p.Args["context"].(string)
 	if ok {
+		if graphErr != nil {
+			return nil, graphConnectionErr()
+		}
 		entities, err := graph.GetRelationships(id, context)
 		if err != nil {
 			return nil, err
@@ -92,6 +95,9 @@ func createEntityHandler() (*handler.Handler, error) {
 			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
 				id, ok := p.Args["id"].(string)
 				if ok {
+					if graphErr != nil {
+						return nil, graphConnectionErr()
+					}
 					entity, err := graph.GetEntity(id)
 					if err != nil {
 						return nil, err
@@ -105,6 +111,9 @@ func createEntityHandler() (*handler.Handler, error) {
 			Type:        nonNullEntityListType,
 			Description: "Get list of Entities",
 			Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+				if graphErr != nil {
+					return nil, graphConnectionErr()
+				}
 				return graph.ListEntities()
 			},
 		},

--- a/app/main.go
+++ b/app/main.go
@@ -8,9 +8,9 @@ import (
 )
 
 var (
-	conf  Config
-	graph database.Graph
-	graphErr   error
+	conf     Config
+	graph    database.Graph
+	graphErr error
 )
 
 func main() {

--- a/app/main.go
+++ b/app/main.go
@@ -10,15 +10,15 @@ import (
 var (
 	conf  Config
 	graph database.Graph
-	err   error
+	graphErr   error
 )
 
 func main() {
 	conf.getConf()
 
-	graph, err = database.GetGraph(conf.Database.Type, conf.Database.Endpoint)
-	if err != nil {
-		log.Println(err)
+	graph, graphErr = database.GetGraph(conf.Database.Type, conf.Database.Endpoint)
+	if graphErr != nil {
+		log.Println(graphErr)
 	}
 
 	dh, err := createEntityHandler()
@@ -32,10 +32,6 @@ func main() {
 		log.Println(err)
 	}
 	http.Handle("/admin", disableCors(ah))
-
-	if err = reIndex(graph, conf.Lakes); err != nil {
-		log.Println(err)
-	}
 
 	log.Printf("Launching Orbital API at http://127.0.0.1:%d\n", conf.Service.Port)
 	if err = http.ListenAndServe(fmt.Sprintf(":%d", conf.Service.Port), nil); err != nil {

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -14,7 +14,7 @@ services:
       - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
       - AWS_REGION=${AWS_REGION}
     ports:
-      - 5000:5000
+      - "5000:5000"
 #  ui:
 #    build: ui
 #    ports:

